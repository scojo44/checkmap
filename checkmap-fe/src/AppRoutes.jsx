import React from 'react'
import {Routes, Route} from 'react-router-dom'
import Map from './Map'
import UserRoutes from './UserRoutes'
import Landing from './Landing'
import LoginForm from './user/LoginForm'
import SignupForm from './user/SignupForm'
import ProfileForm from './user/ProfileForm'
import ListManager from './lists/ListManager'
import CreateListForm from './lists/CreateListForm'

export default function AppRoutes({login, signup, updateUser, addNewList, closeModal}) {
  return (
    <Routes>
      <Route element={<Map/>}>
        <Route path="/" element={<Landing/>}/>
        <Route path="/login" element={<LoginForm login={login}/>}/>
        <Route path="/signup" element={<SignupForm signup={signup}/>}/>
        <Route element={<UserRoutes/>}>
          <Route path="/lists" element={<ListManager {...{closeModal}}/>}/>
          <Route path="/newlist" element={<CreateListForm {...{addNewList, closeModal}}/>}/>
          <Route path="/profile" element={<ProfileForm {...{updateUser, closeModal}}/>}/>
        </Route>
      </Route>
    </Routes>
  );
}
