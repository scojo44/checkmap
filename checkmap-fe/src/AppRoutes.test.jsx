import {render} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import ReactModal from 'react-modal'
import UserContext from './UserContext'
import AppRoutes from './AppRoutes'

const list = {
  id: 1,
  name: 'Test List 1a',
  description: 'This is a test list.',
  color: '#0000FF',
  regionType: 'State',
  ownerName: 'u1',
  states: [],
  counties: []
}
const user = {
  username: 'testuser',
  imageURL: 'u1.jpeg',
  role: 'User',
  lists: [list]
}
const context = {
  user,
  currentList: list,
  setCurrentList: x => x,
  showAlert: x => x
};

describe('AppRoutes Tests', () => {
  // React-Modal expects a top-level element with ID of 'root' for accessibility
  // Adding a <div id="root"> around the <ModalOutlet> doesn't work.
  ReactModal.setAppElement = vi.fn();

  it('Renders without crashing', () => {
    render(
      <MemoryRouter>
        <UserContext.Provider value={context}>
          <AppRoutes />
        </UserContext.Provider>
      </MemoryRouter>
    );
  });

  it('Matches snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <UserContext.Provider value={context}>
          <AppRoutes />
        </UserContext.Provider>
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
