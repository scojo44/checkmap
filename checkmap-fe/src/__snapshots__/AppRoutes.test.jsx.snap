// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppRoutes Tests > Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="leaflet-container leaflet-touch leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
    style="position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      >
        <div
          class="leaflet-layer "
          style="z-index: 1;"
        >
          <div
            class="leaflet-tile-container leaflet-zoom-animated"
            style="z-index: 18; left: 0px; top: 0px;"
          >
            <img
              alt=""
              class="leaflet-tile"
              src="https://b.tile.openstreetmap.org/5/7/12.png"
              style="width: 256px; height: 256px; left: -119px; top: -29px;"
            />
          </div>
        </div>
      </div>
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      />
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      >
        <div
          class="leaflet-control-zoom leaflet-bar leaflet-control"
        >
          <a
            aria-disabled="false"
            aria-label="Zoom in"
            class="leaflet-control-zoom-in"
            href="#"
            role="button"
            title="Zoom in"
          >
            <span
              aria-hidden="true"
            >
              +
            </span>
          </a>
          <a
            aria-disabled="false"
            aria-label="Zoom out"
            class="leaflet-control-zoom-out"
            href="#"
            role="button"
            title="Zoom out"
          >
            <span
              aria-hidden="true"
            >
              −
            </span>
          </a>
        </div>
      </div>
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JavaScript library for interactive maps"
          >
            Leaflet
          </a>
           
          <span
            aria-hidden="true"
          >
            |
          </span>
           © 
          <a
            href="https://www.openstreetmap.org/copyright"
          >
            OpenStreetMap
          </a>
           contributors
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
